Firestore/core/src/firebase/firestore/remote/connectivity_monitor_apple.mm

```diff
@@ -90,9 +92,11 @@ class ConnectivityMonitorApple : public ConnectivityMonitor {
       return;
     }
 
-    auto queue =
-        dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0);
-    success = SCNetworkReachabilitySetDispatchQueue(reachability_, queue);
+    // On Apple platforms, the executor implementation must be the
+    // libdispatch-based one.
+    auto executor = static_cast<ExecutorLibdispatch*>(queue()->executor());
```

Comment:
This cast makes me uncomfortable because this requirement is not generally true. It only happens to be in this case. I can't come up with a great way to resolve this though because we want a generic way to create these things :-(.

Would it be worth adding a type() method and an enum to Executor so that we can make an assertion here? (If you think that's too heavy handed, I'm ok with that too.